src/main/java/org/openstreetmap/atlas/streaming/resource/zip/ZipFileWritableResource.java

```diff
@@ -44,64 +29,38 @@ public ZipFileWritableResource(final File source)
     {
-        try (ZipFile file = new ZipFile(getFileSource().getFile())
+        try
         {
-            @Override
-            public void close()
+            final ZipInputStream iteratorStream = getZipInputStream();
+            ZipEntry currentZipEntry;
+            final List<Resource> resources = new ArrayList<>();
+            while ((currentZipEntry = iteratorStream.getNextEntry()) != null)
             {
-                // Do nothing to close the file here, to avoid cutting the legs off the just created
-                // ZipEntry-based resource.
+                final String entryName = currentZipEntry.getName();
+                resources.add(new InputStreamResource(() ->
+                {
+                    final ZipInputStream zipStream = getZipInputStream();
+                    seekStreamAheadToEntry(zipStream, entryName);
+                    return zipStream;
+                }).withName(entryName));
             }
-        })
-        {
-            return Iterables.translate(Iterables.from(file.entries()), entry ->
-            {
-                return new InputStreamResource(inputStreamSupplier(file, entry))
-                        .withName(entry.getName());
-            });
+            return resources;
         }
-        catch (final IOException e)
+        catch (final IOException exception)
         {
-            throw new CoreException("Cannot get entries from the Zipfile {}.",
-                    this.getFileSource().getName(), e);
+            throw new CoreException("Could not read entries for {}",
+                    getFileSource().getAbsolutePathString(), exception);
         }
     }
 
-    /**
-     * Jump to a specific entry.
-     *
-     * @param name
-     *            The name of the entry
-     * @return The entry as a {@link Resource}. Throws a {@link CoreException} if it cannot find the
-     *         entry.
-     */
     public Resource entryForName(final String name)
     {
-        try (ZipFile file = new ZipFile(getFileSource().getFile())
-        {
-            @Override
-            public void close()
-            {
-                // Do nothing to close the file here, to avoid cutting the legs off the just created
-                // ZipEntry-based resource.
-            }
-        })
-        {
-            final ZipEntry entry = file.getEntry(name);
-            if (entry != null)
-            {
-                return new InputStreamResource(inputStreamSupplier(file, entry)).withName(name);
-            }
-            else
-            {
-                throw new IOException("Entry " + name + " does not exist.");
-            }
-        }
-        catch (final IOException e)
+        return new InputStreamResource(() ->
         {
-            throw new CoreException("Cannot get the entry {} from the Zipfile {}.", name,
-                    this.getFileSource().getName(), e);
-        }
+            final ZipInputStream zipStream = getZipInputStream();
```

Comment:
Are there any performance differences with this approach, i.e. opening a new file and its stream for each entry? Not sure if the `streamToSeekThrough.getNextEntry();` below is expensive or not.

Reply:
There is a moderate performance hit. I ran some tests where I rapidly opened a bunch of different atlas fields. Nothing huge. For example, opening and atlas and reading all the edge identifiers takes on average ~50ms longer now.